db/controllers/comments.controller.js

```diff
@@ -1,15 +1,42 @@
-const { fetchCommentsByArticle } = require('../models/comments.model')
+const { fetchCommentsByArticle, postNewComment } = require('../models/comments.model')
+const { doesArticleExist } = require('../models/articles.model')
 
 exports.getCommentsByArticle = (req, res, next) => {
     const articleId = +req.params.article_id;
     if (isNaN(articleId)){
         return res.status(400).send({ message: "ID must be a number" });
     }
-    fetchCommentsByArticle(articleId).then((comments) => {
-        if (comments.length === 0){
-            return Promise.reject({ status: 404, message: "invalid article ID"})
+    Promise.all([doesArticleExist(articleId), fetchCommentsByArticle(articleId)])
+    .then(([articleExists, comments]) => {
+        if (!articleExists) {
+        res.status(404).send({ message: `Article not found for article ID: ${articleId}` });
+    } else {
+        res.status(200).send({ comments });
+    }
+    })
+    .catch(next); 
+};
+
+exports.addNewComment = (req, res, next) => {
+    const articleId = +req.params.article_id;
+    if (Object.keys(req.body).length < 2) {
+        res.status(400).send({ err: 400, msg: 'Comment must contain username and body' });
+        }
+    const { username, body } = req.body;
+    if (body.length === 0){
+        res.status(400).send({ err: 400, msg: 'Comment cannot be blank'})
+    }
```

Comment:
A lot of your error logic is being handled in the controller. Ideally, this should be in the model so that the controller can just fall into your catch block and invoke the error handlers you have set up in app.js